File: sdks/csharp/Examples/SpriteDemo.cs
/**
 * 2D Sprite Demo
 * 
 * Demonstrates 2D sprite rendering with the Windjammer C# SDK.
 * 
 * Run with: dotnet run --project Examples/SpriteDemo.cs
 */

using System;
using Windjammer.SDK;

namespace Windjammer.Examples
{
    class SpriteDemo
    {
        static void Main(string[] args)
        {
            Console.WriteLine("=== Windjammer 2D Sprite Demo (C#) ===");

            // Create 2D application
            var app = new App();

            // Setup system
            app.AddStartupSystem(() =>
            {
                Console.WriteLine("\n[Setup] Creating 2D scene...");

                // Create camera
                var camera = new Camera2D(new Vector2(0, 0), 1.0f);
                Console.WriteLine($"  - {camera}");

                // Create sprites
                var sprite1 = new Sprite
                {
                    Texture = "player.png",
                    Position = new Vector2(0, 0),
                    Size = new Vector2(64, 64)
                };
                Console.WriteLine("  - Sprite 'player.png' at (0, 0) size=(64, 64)");

                var sprite2 = new Sprite
                {
                    Texture = "enemy.png",
                    Position = new Vector2(100, 100),
                    Size = new Vector2(48, 48)
                };
                Console.WriteLine("  - Sprite 'enemy.png' at (100, 100) size=(48, 48)");

                Console.WriteLine("[Setup] Scene ready!");
            });

            // Update system
            app.AddSystem(() =>
            {
                // This would rotate sprites each frame
            });

            Console.WriteLine("2D application configured!");
            Console.WriteLine("- Camera: Orthographic");
            Console.WriteLine("- Sprites: Enabled");
            Console.WriteLine("- Physics: 2D");
            Console.WriteLine();

            // Run the application
            app.Run();
        }
    }
}

File: sdks/kotlin/examples/SpriteDemo.kt
/**
 * 2D Sprite Demo
 * 
 * Demonstrates 2D sprite rendering with the Windjammer Kotlin SDK.
 * 
 * Run with: gradle run -PmainClass=dev.windjammer.examples.SpriteDemo
 */

package dev.windjammer.examples

import dev.windjammer.sdk.*

fun main() {
    println("=== Windjammer 2D Sprite Demo (Kotlin) ===")

    // Create 2D application
    val app = App()

    // Setup system
    app.addStartupSystem {
        println("\n[Setup] Creating 2D scene...")

        // Create camera
        val camera = Camera2D(Vec2(0f, 0f), 1.0f)
        println("  - $camera")

        // Create sprites
        val sprite1 = Sprite(
            texture = "player.png",
            position = Vec2(0f, 0f),
            size = Vec2(64f, 64f)
        )
        println("  - Sprite 'player.png' at (0, 0) size=(64, 64)")

        val sprite2 = Sprite(
            texture = "enemy.png",
            position = Vec2(100f, 100f),
            size = Vec2(48f, 48f)
        )
        println("  - Sprite 'enemy.png' at (100, 100) size=(48, 48)")

        println("[Setup] Scene ready!")
    }

    // Update system
    app.addSystem {
        // This would rotate sprites each frame
    }

    println("2D application configured!")
    println("- Camera: Orthographic")
    println("- Sprites: Enabled")
    println("- Physics: 2D")
    println()

    // Run the application
    app.run()
}

